src/richie/apps/courses/templates/courses/cms/course_detail.html

```diff
@@ -368,22 +368,28 @@ <h2 class="course-detail__title">{% blocktrans context "course_detail__title" %}
             {% block runs %}
                 <div class="course-detail__aside">
                     {% render_model_add course "" "" "get_admin_url_to_add_run" %}
-                    {% with runs_dict=current_page.course.course_runs_dict %}
-                        {% with open_visible_runs=runs_dict.0|add:runs_dict.1|add:runs_dict.2|visible_on_course_page:request.toolbar.edit_mode_active %}
-                            <!--
-                                As course runs are rendered through React, we use meta tags to
-                                setup course runs RFDa properties.
-                             -->
-                            {% for run in open_visible_runs %}
-                                <span rel="hasCourseInstance" typeof="CourseInstance">
-                                    <meta property="name" content="{{ run.title }}"  />
-                                    <meta property="inLanguage" content="{{ run.get_languages_display }}"  />
-                                    <meta property="courseMode" content="online"  />
-                                    <meta property="startDate" content="{{ run.start|date:'Y-m-d' }}"  />
-                                    <meta property="endDate" content="{{ run.end|date:'Y-m-d' }}"  />
+                    {% with runs=current_page.course.course_runs|visible_on_course_page:request.toolbar.edit_mode_active %}
+                        <!--
+                            As course runs are rendered through React, we use meta tags to
+                            setup course runs RFDa properties.
+                            -->
+                        <span rel="offers" typeof="Offer">
+                            <meta property="category" content="Partially Free" />
```

Comment:
It needs to have a value of:
- Free: All lectures, assignments, and tests must be available without cost for all users. The entire course can be completed without cost.
- Partially Free: More than half of the course can be completed for free (for example, all lectures are free of charge, but assignments are paid).
- Subscription: The user must be a subscriber or paid member in order to complete the entire course.
- Paid: The user must pay to complete the course.

Reference: https://developers.google.com/search/docs/appearance/structured-data/course-info#guidelines

But the main problem is that it depends on the LMS backend of the course. Do you feel that we should add it as an option for each backend implementation? On the React code like: `rdfa_offers`.
- Open edX - MOOC like, using `Partially Free`, in future it could query LMS API for the course modes and its price;
- Joanie other business logic.

Example of dictionary:
```
   "category": "Partially Free",
```

```
   "category": "Paid",
   "priceCurrency": "EUR",
   "price": 10.99
```
Reference: https://developers.google.com/search/docs/appearance/structured-data/course-info#guidelines

Reply:
@jbpenrath what about if I add a new Django setting and show the `offers.category` only if it is defined? Alternatively I can add the `offers` on an override of the `runs` block.

Reply:
With our paid courses, we plan to improve CourseRun model to add a field to distinguish paid course of free courses. I feel our needs meets your needs on this point. What do you think to add a field `offers` into the CourseRun model (by default it could set to `free`)?

Reply:
@jbpenrath I think it goes into the right direction!!

Nevertheless, I think that value should be on the `CourseRun` level instead. That value could change overtime, and a good way to change that value on a business process (and our relation with partners), is to change it for a new course run.
About the `offers` field. Yes, by default its value could be set to `free`, in reality the default value should be read from a setting.
I think we should also include the `priceCurrency` and `price` as new fields to `CourseRun`. On RDFa the `Course` `offers` should be only included for opened for enrollment courses.

Additionally, make those values editable also using the course runs sync API, the `CouseRun` fields:
- `offers` enumeration of: `Free`, `Partially Free`, `Subscription` and `Paid`
- priceCurrency - `The currency of the price of the award, in [ISO 4217 currency format](https://wikipedia.org/wiki/ISO_4217#List_of_ISO_4217_currency_codes) (3 letter code)`
- price

With this features, it could be possible to change the layout of edx backend course (not the Joanie backend courses) to include that `offers` information. If it is a completely `free` course, `Partially Free` a M**O**OC, or your training `Subscription` and `Paid` courses.

Further stuff, on the specification there is also the `educationalCredentialAwarded`. But I think we don't need it, and at least NAU won't use that business model - pay only to obtain the certificate.

FYI @sandroscosta

Reply:
I'm going to remove the block.
I'm going to override on our site using the `block runs`, at least until we don't have a it configurable from the UI.
```html
<span rel="offers" typeof="Offer">
    <meta property="category" content="Partially Free" />
</span>
```

@jbpenrath do you agree with my previous comment ? https://github.com/openfun/richie/pull/2445#discussion_r1654843549

If yes, I could create a new Issue so we better track what needs to be implemented.
The even further step could be create search filters with `offers` and `price`.

Reply:
Yes, your comment looks fine for me. It looks a good idea to open an issue to backup our thoughts on that.